pkg/common/util/dial.go

```diff
@@ -0,0 +1,79 @@
+package util
+
+import (
+	"context"
+	"net"
+	"time"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+// BlockingDial is a helper method to dial the given address, using optional TLS credentials,
+// and blocking until the returned connection is ready. If the given credentials are nil, the
+// connection will be insecure (plain-text).
+// (This function was borrowed from https://github.com/fullstorydev/grpcurl/blob/master/grpcurl.go)
```

Comment:
If we are going to use this function as-is (see other comments), then we need to include the MIT license in this file: https://github.com/fullstorydev/grpcurl/blob/master/LICENSE